jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/Booster.java

```diff
@@ -43,10 +57,29 @@
    *                  the prediction of these DMatrices will become faster than not-cached data.
    * @throws XGBoostError native error
    */
-  Booster(Map<String, Object> params, DMatrix[] cacheMats) throws XGBoostError {
+  static Booster newBooster(Map<String, Object> params, DMatrix[] cacheMats) throws XGBoostError {
+    if (USE_KRYO_BOOSTER)
```

Comment:
Does it make sense to make the check lazy via Holder class (aka https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom)?  My worry is mainly about side-effects - if we try to initialize Kryo class if it triggers any other static initializers.

Reply:
Should not be a concern imho - doesn't change the previous behavior.